linfa-ensemble/src/random_forest/algorithm.rs

```diff
@@ -0,0 +1,122 @@
+use linfa_trees::DecisionTree;
+use crate::random_forest::hyperparameters::RandomForestParams;
+use ndarray::{Array1, Array2, ArrayBase, Data, Ix1, Ix2};
+use ndarray_rand::rand_distr::Uniform;
+use ndarray::Axis;
+use ndarray::Array;
+use ndarray_rand::RandomExt;
+use std::collections::HashMap;
+
+
+pub struct RandomForest {
+    pub hyperparameters: RandomForestParams,
+    pub trees: Vec<DecisionTree>,
+}
+
+impl RandomForest {
+    pub fn fit(
+        hyperparameters: RandomForestParams,
+        x: &ArrayBase<impl Data<Elem = f64>, Ix2>,
+        y: &ArrayBase<impl Data<Elem = u64>, Ix1>,
+    ) -> Self {
+
+    let n_estimators = hyperparameters.n_estimators;
+    let mut trees: Vec<DecisionTree> = Vec::with_capacity(n_estimators);
+    let single_tree_params = hyperparameters.tree_hyperparameters;
+
+    //TODO check bootstrap
+    let _bootstrap = hyperparameters.bootstrap;
```

Comment:
this parameter controls whether to sample with replacement or without?

Reply:
yes. Just a placeholder now as functionality not yet implemented.